### ISSUE TITLE
[HTTP/3] UriFormatException when Alt-Svc omits host

### ISSUE THREAD

**Issue by wegylexy** [dotnet/runtime#53632]
### Description
The `Alt-Svc` header from a HTTP/3-enabled site may look like `h3-27=":443"; ma=86400, h3-28=":443"; ma=86400, h3-29=":443"; ma=86400, h3=":443"; ma=86400`, without specifying the host.
In this case, the original host should be assumed. However, `HttpConnectionPool` simply passes `null` to the `HttpAuthority` constructor which directly feeds it into a `UriBuilder`, resulting in a `UriFormatException`.
It should use the same host of the original HTTP request when omitted in `Alt-Svc`.

### Regression?

It would work before 6f6c94dbb2259deb083eefb755698cc8e0f0a4ce but the point is to properly support HTTP/3, which was draft 34 when this issue was filed.

For a temporary workaround, add the following line before the HTTP connection pool is created to disable HTTP/3 draft support.
```cs
AppContext.SetSwitch("System.Net.SocketsHttpHandler.Http3DraftSupport", false);
```

### Other information

Since initial HTTP/3 support:
https://github.com/dotnet/runtime/blob/6f6c94dbb2259deb083eefb755698cc8e0f0a4ce/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs#L800
When this issue was filed:
https://github.com/dotnet/runtime/blob/ad6a69735d61a1bc22da9aff15ebc5df94f63beb/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs#L1007
Where `UriFormatException` is thrown:
https://github.com/dotnet/runtime/blob/ad6a69735d61a1bc22da9aff15ebc5df94f63beb/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpAuthority.cs#L23

**Comment by ManickaP** [comment]
Triage: we should fix that for 6.0, this is a part of good H/3 story. Thanks for bringing this up.

**Comment by wegylexy** [comment]
Ideally, it should just work with older HTTP versions when upgrade fails for whatever reason, without throwing an exception.
